### DISCUSSION TITLE
I need help understanding what this LNK2019 error means

### DISCUSSION THREAD

**Discussion by Ashish308** [microsoft/AirSim#4719] [Support Q&A]
Build started...
1>------ Build started: Project: WildfireAirSim, Configuration: DebugGame_Editor x64 ------
2>------ Skipped Build: Project: UE4, Configuration: BuiltWithUnrealBuildTool Win32 ------
2>Project not selected to build for this solution configuration 
1>Using 'git status' to determine working set for adaptive non-unity build (C:\Users\Ashis\source\repos\WildfireAirSim).
1>Waiting for 'git status' command to complete
1>Building WildfireAirSimEditor...
1>Using Visual Studio 2022 14.33.31630 toolchain (C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.33.31629) and Windows 10.0.19041.0 SDK (C:\Program Files (x86)\Windows Kits\10).
1>Building 2 actions with 20 processes...
1>  [1/2] UE4Editor-WildfireAirSim-Win64-DebugGame.dll
1>     Creating library C:\Users\Ashis\source\repos\WildfireAirSim\Intermediate\Build\Win64\UE4Editor\DebugGame\WildfireAirSim\UE4Editor-WildfireAirSim-Win64-DebugGame.suppressed.lib and object C:\Users\Ashis\source\repos\WildfireAirSim\Intermediate\Build\Win64\UE4Editor\DebugGame\WildfireAirSim\UE4Editor-WildfireAirSim-Win64-DebugGame.suppressed.exp
1>WAHUD.cpp.obj : error LNK2019: unresolved external symbol "public: static void __cdecl UAirBlueprintLib::OnBeginPlay(void)" (?OnBeginPlay@UAirBlueprintLib@@SAXXZ) referenced in function "public: void __cdecl AWAHUD::OnResetLevelAfter(void)" (?OnResetLevelAfter@AWAHUD@@QEAAXXZ)
1>WAHUD.cpp.obj : error LNK2019: unresolved external symbol "public: static void __cdecl UAirBlueprintLib::OnEndPlay(void)" (?OnEndPlay@UAirBlueprintLib@@SAXXZ) referenced in function "public: void __cdecl AWAHUD::OnResetLevelBefore(void)" (?OnResetLevelBefore@AWAHUD@@QEAAXXZ)
1>C:\Users\Ashis\source\repos\WildfireAirSim\Binaries\Win64\UE4Editor-WildfireAirSim-Win64-DebugGame.dll : fatal error LNK1120: 2 unresolved externals
1>C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.MakeFile.Targets(44,5): error MSB3073: The command ""C:\Program Files\Epic Games\UE_4.27\Engine\Build\BatchFiles\Build.bat" WildfireAirSimEditor Win64 DebugGame -Project="C:\Users\Ashis\source\repos\WildfireAirSim\WildfireAirSim.uproject" -WaitMutex -FromMsBuild" exited with code 6.
1>Done building project "WildfireAirSim.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 1 skipped ==========

**Comment by cnguyen770** [comment]
I believe your problem is that you are missing AirSim dependencies. Your fix would be to uninstall and reinstall AirSim.
